Convert project to a CLI tool

Oneill19 opened the pull request:
## Overview
This pull request transforms the existing `ng-stats` project into a CLI tool. This change allows users to run `ng-stats` directly from the command line.

## Changes Made
- **Changed the Path Handling:** Updated the tool to use the current working directory (`cwd`) to determine where to execute, making it more flexible when run from different project directories.
- **Project Validation:** Added a validation step to check for the presence of an `angular.json` file in the directory. This ensures that the tool is run in the correct context and prevents errors from occurring in non-Angular projects.
- **Updated Documentation:** Revised the README to reflect the new CLI usage instructions, providing users with clear guidance on how to use the tool in its new form.
- **Build Command Addition:** Added a new build command in `package.json` to streamline the setup process for new users and ensure that the tool can be easily built from source.

Oneill19 committed:
convert to cli-tool

Oneill19 committed:
update README.md

tomer953 commented:
Awesome! thanks for your contribution! 🚀🚀

However, the `angular.json` validation part is a bit problem, since in case of a [workspace](https://angular.io/guide/file-structure) the sub projects does not include `angular.json` for each folder.

so in case I want to run the tool for `projects/project-a` only, and not for the whole workspace, I can't with this validation.


For now lets give the consumer the option to run with the currentt folder, or for a specific path:

with: `--path /users/foo/bar/projects/a`

see:
https://www.npmjs.com/package/minimist

Oneill19 committed:
add minimist

Oneill19 committed:
add cli options and update readme

Oneill19 commented:
Thank you for the valuable feedback! 🚀

I've updated the CLI tool. Here's what has been done:
1. **Path Flexibility:** Users can now specify a path using the `--path` option. This allows them to direct the tool to analyze any specific subproject within a workspace by providing the exact path to that project.
2. **JSON Output Option:** For better integration with other tools and systems that might consume the output of `ng-stats`, I've added a `--json` option. This allows the output to be in JSON format, which is more suitable for automated parsing and further processing
3. **Help Option:** I've also included a `--help` option to make it easier for new users to understand how to use the tool effectively.

The tool uses the `minimist` package to handle command-line arguments efficiently, making it flexible and user-friendly. Here’s an example command to run the tool for a specific subproject:

```bash
ng-stats --path users/foo/bar/projects/project-a
```

tomer953 commented:
Wow! Awesome job! 💪 Merged
